src/Raven.Client/Http/RequestExecutor.cs

```diff
@@ -627,11 +659,16 @@ private HttpRequestMessage CreateRequest<TResult>(ServerNode node, RavenCommand<
 
         public event Action<StringBuilder> AdditionalErrorInformation;
 
-        private async Task<bool> HandleUnsuccessfulResponse<TResult>(ServerNode chosenNode, int nodeIndex, JsonOperationContext context, RavenCommand<TResult> command, HttpRequestMessage request, HttpResponseMessage response, string url)
+        private async Task<bool> HandleUnsuccessfulResponse<TResult>(ServerNode chosenNode, int? nodeIndex, JsonOperationContext context, RavenCommand<TResult> command, HttpRequestMessage request, HttpResponseMessage response, string url, int? sessionId)
         {
             switch (response.StatusCode)
             {
                 case HttpStatusCode.NotFound:
+                    if (nodeIndex.HasValue)
```

Comment:
This will cause loading a non existing document (which is 404) to mark the node as failure.